examples/basic.zig

```diff
@@ -22,26 +10,27 @@ fn afterAllHook() void {
 }
 
 fn myBenchmark(_: std.mem.Allocator) void {
-    _ = helloWorld();
+    var result: usize = 0;
+    for (0..1_000_000) |i| {
+        std.mem.doNotOptimizeAway(i);
+        result += i * i;
+    }
 }
 
 test "bench test basic" {
     const stdout = std.io.getStdOut().writer();
-    var bench = zbench.Benchmark.init(test_allocator, .{});
+    var bench = zbench.Benchmark.init(std.testing.allocator, .{});
     defer bench.deinit();
 
+    const sysinfo = try bench.getSystemInfo();
+    try std.fmt.format(stdout, "\n{}\n", .{sysinfo});
+
     try bench.add("My Benchmark", myBenchmark, .{
-        .iterations = 10,
+        .iterations = 100,
         .hooks = .{
             .before_all = beforeAllHook,
             .after_all = afterAllHook,
         },
     });
-
-    const sysinfo = try bench.getSystemInfo();
-    try std.fmt.format(stdout, "\n{}\n", .{sysinfo});
-
-    const results = try bench.run();
-    defer results.deinit();
-    try results.prettyPrint(stdout, true);
+    try bench.run(stdout);
```

Comment:
Any objection to doing the pretty printing and handling std.Progress in the `run()` function? I think it's ok for it to do more for the common case when there's the `iterator()` method which gives lower level access.

Reply:
No objections, i think thats justified.